matrix/Quaternion.hpp

```diff
@@ -280,9 +280,11 @@ class Quaternion : public Vector<Type, 4>
     void invert()
     {
         Quaternion &q = *this;
-        q(1) *= -1;
-        q(2) *= -1;
-        q(3) *= -1;
+        Type n = q.dot(q);
```

Comment:
It is over the norm (without square) because the definition of the inverse is:
q * q^-1 = q^-1 * q = q_I    where the identity quaternion q_I = [1 0 0 0] has unit length.
You can infer from the quaternion multiplication that |q^-1| = 1/|q|.

The matlab implementation is over norm and you can also look up a definition somwhere else for example https://www.j3d.org/matrix_faq/matrfaq_latest.html for example says:
```
How do I calculate the inverse of a quaternion?
----------------------------------------------------
  This is equivalent to calculating the conjugate of the quaternion,
  if the quaternion is normalized (or a unit quaternion).
  In all other cases, the magnitude of the inverse is 1/|q|.
```